kedro/contrib/io/networkx_local/networkx_local.py

```diff
@@ -0,0 +1,110 @@
+# (either separately or in combination, "QuantumBlack Trademarks") are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""``NetworkXLocalDataSet`` loads and saves graphs to a local json file in node/link format using
+``NetworkX``.
+See https://networkx.github.io/documentation/stable/tutorial.html for details.
+"""
+
+import json
+from pathlib import Path
+from typing import Any, Dict, Union
+
+import networkx
+
+from kedro.contrib.io import DefaultArgumentsMixIn
+from kedro.io import AbstractDataSet
+from kedro.io.core import DataSetError
+
+
+class NetworkXLocalDataSet(DefaultArgumentsMixIn, AbstractDataSet):
+    """
+    ``NetworkXLocalDataSet`` loads and saves graphs to a local json file in node/link format using
+    ``NetworkX``.
+    See https://networkx.github.io/documentation/stable/tutorial.html for details.
+    Example:
+    ::
+        >>> from kedro.contrib.io.networkx_local import NetworkXLocalDataSet
+        >>> import networkx as nx
+        >>> graph = nx.complete_graph(100)
+        >>> graph_dataset = NetworkXLocalDataSet(filepath="test.json")
+        >>> graph_dataset.save(graph)
+        >>> reloaded = graph_dataset.load()
+        >>> assert nx.is_isomorphic(graph, reloaded)
+    """
+
+    def _describe(self) -> Dict[str, Any]:
+        return dict(
+            filepath=self._filepath,
+            load_args=self._load_args,
+            save_args=self._save_args,
+        )
+
+    def __init__(
+        self,
+        filepath: Union[str, Path],
+        load_args: Dict[str, Any] = None,
+        save_args: Dict[str, Any] = None,
+    ) -> None:
+        """Creates a new instance of ``NetworkXLocalDataSet``.
+
+        Args:
+            filepath: The path to the NetworkX graph JSON file.
+
+            load_args: Arguments passed on to ```networkx.node_link_graph``.
+                See https://networkx.github.io/documentation for details.
+
+            save_args: Arguments passed on to ```networkx.node_link_data``.
+                See https://networkx.github.io/documentation for details.
+
+        """
+        self._filepath = Path(filepath)
+        super().__init__(load_args, save_args)
+
+    def _load(self) -> Union[networkx.Graph]:
+        graph = None
+        if not self._exists():
+            raise DataSetError(
+                "Failed while loading data from data set NetworkXLocalDataSet()"
+            )
+        with self._filepath.open("r") as input_file:
+            json_payload = json.loads(input_file.readline())
```

Comment:
I think `readline()` won't do what is expected because it will only read a single line from the file (the json file could have several lines).
You could do `json_payload = json.loads(self._filepath.read_text())`, and you can delete `with self._filepath.open("r") as input_file:` line.